protocol/packages/swap/src/astroport/mod.rs

```diff
@@ -95,24 +73,35 @@ where
         })
     }
 
-    fn parse_response<I>(trx_resps: &mut I) -> Result<Amount>
-    where
-        I: Iterator<Item = CosmosAny>,
-    {
-        trx_resps
-            .next()
-            .ok_or_else(|| Error::MissingResponse("router swap".into()))
-            .and_then(|resp| {
-                trx::decode_msg_response::<_, ResponseMsg>(resp, ResponseMsg::type_url())
+    fn parse_response(response: CosmosAny) -> Result<Amount> {
+        trx::decode_msg_response::<_, ResponseMsg>(response, ResponseMsg::type_url())
+            .map_err(Into::into)
+            .and_then(|response_message| {
+                cosmwasm_std::from_json::<SwapResponseData>(response_message.data)
                     .map_err(Into::into)
             })
-            .and_then(|cosmwasm_resp| {
-                cosmwasm_std::from_json::<SwapResponseData>(cosmwasm_resp.data).map_err(Into::into)
-            })
             .map(|swap_resp| swap_resp.return_amount.into())
     }
 }
 
+trait Router {
+    const ADDRESS: &'static str;
+}
+
+#[cfg(feature = "dex-astroport_test")]
+impl Router for Impl {
+    // Source: https://github.com/astroport-fi/astroport-changelog/blob/main/neutron/pion-1/core_testnet.json
+    const ADDRESS: &'static str =
+        "neutron12jm24l9lr9cupufqjuxpdjnnweana4h66tsx5cl800mke26td26sq7m05p";
+}
+
+#[cfg(feature = "dex-astroport_main")]
+impl Router for Impl {
+    // Source: https://github.com/astroport-fi/astroport-changelog/blob/main/neutron/neutron-1/core_mainnet.json
+    const ADDRESS: &'static str =
+        "neutron1rwj6mfxzzrwskur73v326xwuff52vygqk73lr7azkehnfzz5f5wskwekf4";
+}
+
```

Comment:
The more I look at this more I doubt the trait is needed. To say it straight - it's an overkill. In addition, the new way you suggest to be consumed is not acceptable - `<Self as Router>`

Reply:
Why wouldn't it be acceptable? And I wouldn't say it's an overkill as this **dis**allows different signatures while also evading `cfg` shenanigans in the constant's value definition.